internal/provider-integration/v2/command_create.go

```diff
@@ -68,10 +68,16 @@ func (c *command) create(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("cloud provider must be either %s or %s", providerAzure, providerGcp)
 	}
 
-	environmentId, err := c.Context.EnvironmentId()
+	environmentId, err := cmd.Flags().GetString("environment")
 	if err != nil {
 		return err
 	}
+	if environmentId == "" {
+		environmentId, err = c.Context.EnvironmentId()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
We should revert this since `c.Context.EnvironmentId()` will already grab the flag value if specified.

That's because the CLI temporarily overwrites the current environment with the user provided value in the prerunner.